api/pullrequests/pr.controller.js

```diff
@@ -0,0 +1,141 @@
+			});
+		},
+
+    getOnePr: (req, res, done) => {
+      // To get a single PR
+      const { id } = req.params;
+      PullRequest.findById(id).exec((err, pr) => {
+        if (err) return util.errorHandler(500, 'Server error locating this pull request.', res, done);
+        return pr ? res.json(pr) : util.errorHandler(404, 'This Pull Request does not exist in the database.', res, done);
+      })
+    },
+
+    deleteOnePr: (req, res, done) => {
+      const { id } = req.params;
+      PullRequest.findByIdAndRemove(id, err => err ? util.errorHandler(500, 'Server error trying to delete this pull request.', res, done) : res.json(response));
+    },
+
+    clearClosedPrs : (req, res, done) => {
+			PullRequest.remove({ state: 'closed'}, err => err ? util.errorHandler(500, 'Server error clearing closed PRs.', res, done) : res.status(200).send({ message: 'All closed PRs have been deleted.' }));
+    },
+    
+    // Whenever somebody goes in a file in a pull request and creates, edits, or deletes a comment for a 'dif'
+    // Should be refactored in the future to be easier to navigate and probably only need to write one call to db.
+    handlePrDiffComment: (req, res, done) => {
+      const action = req.body.action;
+      const prId = req.body.pull_request.id;
+      const ghId = req.body.comment.id;
+      const postComment = {
+        ghId,
+        body: req.body.comment.body,
+        created: new Date(req.body.comment.created_at),
+        file: req.body.comment.path,
+        user: {
+          login: req.body.comment.user.login,
+          avatar_url: req.body.comment.user.avatar_url
+        }
+      }
+
+      const deleteDiffComment = () => {
+        PullRequest.update({id: prId}, 
+          { $pull: { diffComments: { ghId }}}, 
+          { safe: true, multi: true }, 
+          (err, response) => {
+            return err ? util.errorHandler(500, 'Server error deleting this comment from the pull request.', res, done) : res.json(response);
+          })
+      }
+      const createDiffComment = () => {
+        PullRequest.findOne({id: prId})
+          .exec((err, pr) => {
+            if (err) {
+              return util.errorHandler(500, 'Server Error looking up the Pull request to inser a new Diff comment.', res, done);
+            }
+            pr.diffComments.push(postComment);
+            pr.save((err,result) => {
+              return err ? util.errorHandler(500, 'Server saving this diff comment in the pull request', res, done) : res.json(result);
+            })
+          })
+      }
+      const editDiffComment = () => {
+        PullRequest.update({id: prId, 'diffComments': {'$elemMatch': {ghId } }}, 
+          { $set : postComment }, 
+          { new:true, upsert:true, safe:true }, 
+          (err, response) => {
+            return err ? util.errorHandler(500, 'Error updating a diff comment for the pull request', res, done) : res.json(response);
+          })
+      }
+
+      switch(action) {
```

Comment:
Do these actions come from the Hook event submitted from Github?!?! How have I not seen this!?

Reply:
Yes sir, I figured they had to have a way to help us decipher what is going on.lol